Molecular Consequence Schema and Tests

Mrinal-Thomas-Epic opened the pull request:
Initial draft of schema for representing molecular consequence predictions from VEP (and potentially other sources in the future). Also includes a minor refactor to use a common `ConceptSet` class, rather than defining different grouper classes for therapies, conditions, molecular consequence terms, etc.

Note: `ConceptSet` will later be moved into gks-core, but that won't change the final generated schema.


Out of scope:
-  Handling intronic variants. This will be added in a later time, once VRS issue [#650](https://github.com/ga4gh/vrs/issues/650) (actively being worked on) is resolved

larrybabb commented on schema/va-spec/base/va-core-source.yaml:
```diff
@@ -1079,3 +1079,110 @@ $defs:
+        const: "MolecularConsequence"
+        default: "MolecularConsequence"
+        description: MUST be "MolecularConsequence".
+      focusVariant:
```
@Mrinal-Thomas-Epic if the only type of variant for this is `Allele` then we should consider calling it `focusAllele` like we did in the `CohortAlleleFrequencyStudyResult`

larrybabb replied:
and the description should specify 'allele' instead of 'variant'

Mrinal-Thomas-Epic replied:
From the [VEP documentation](https://www.ensembl.org/info/docs/tools/vep/script/vep_example.html#StructVar), it seems like they support annotating adjacencies as well. I'll try to get an example of this

larrybabb commented on schema/va-spec/base/va-core-source.yaml:
```diff
@@ -1079,3 +1079,110 @@ $defs:
         extends: sourceDataSet
         description: >-
           The full data set that provided the reported the functional impact score.
+  MolecularConsequence:
```
I'm fine with this name, but if we stick with it then we should consider renaming the other `XxxStudyResult` base profiles `Xxx` for consistency (and apply the same to all inherited classes).

larrybabb commented on schema/va-spec/base/va-core-source.yaml:
```diff
@@ -1079,3 +1079,110 @@ $defs:
+          - $refCurie: vrs:Allele
+          - $refCurie: gks.core:iriReference
+        description: The variant which this annotation applies to.
+      transcriptMolecule:
```
@ahwagner it would be helpful if there was a validation mechanism to assure that only transcript sequences could be used in this attribute. Wondering if you feel the same and whether or not we should do something about that now.

larrybabb commented on schema/va-spec/base/va-core-source.yaml:
```diff
@@ -1079,3 +1079,110 @@ $defs:
+        oneOf:
+          - $refCurie: vrs:SequenceReference
+          - $refCurie: gks.core:iriReference
+      transcriptVariant:
```
@Mrinal-Thomas-Epic same as above re: `focusAllele` vs `focusVariant`.

larrybabb commented on schema/va-spec/base/va-core-source.yaml:
```diff
@@ -1079,3 +1079,110 @@ $defs:
+        oneOf:
+          - $refCurie: vrs:SequenceReference
+          - $refCurie: gks.core:iriReference
+      proteinVariant:
```
@Mrinal-Thomas-Epic same as above re: `focusAllele` vs `focusVariant`

larrybabb reviewed (changes requested):
I've only gone through the MolecularConsequence piece thus far. I think I'd like to chat asap to make sure I understand the full use cases. I'm not 100% sure why the proteinMolecule and proteinAllele are included as I assume they will only ever be included if they are associated with a transcriptMolecule and transcriptAllele. If that is correct then I think we need to consider a structure that represents that relationship a bit more clearly.  If not, then I need to understand that use case.  

@ahwagner Since I am assuming that the reason we would not include transcriptMolecule and/or transcriptAllele is because of intronic c. alleles (as they don't actually reside on a transcript.  I think this is very clearly indicating VRS NEEDS a solution to showing variants with relationships between underlying sequences of different molecular types (e.g. genomic to transcript, etc...). I'm not super keen on skirting past that issue since intronic c. represented alleles are fundamental to this kind of study result.  We should discuss this a bit to see if there's a good solution that we can propose to VRS 2.1?

larrybabb commented:
@Mrinal-Thomas-Epic If it's not too much trouble, I would love it if you could create separate PRs for the InSilico stuff and the MoleConsq stuff. If that's too difficult let me know.

larrybabb commented:
@ahwagner @Mrinal-Thomas-Epic maybe we should find a block of time asap to draft the intronic vrs class we need to support some of these use cases. we can do this with a small group like the 3 of us and try it out in these va-spec statements before bringing it to the larger group. And since it's draft we can add it to VRS so that EPIC and VEP can move forward with these statements.

mbrush commented:
My primary comment on this PR is that Molecular Consequence really should be a Statement, not a Study Result. Sara Hunt raised this point on the last VA Community Call, and I strongly concur.
				
Some additional notes about the rationale for this and a proposed model are in the document [here](https://docs.google.com/document/d/1DWRPjopZcVpyp1zMvjh7J3yplUIJc3R4E26COJok6Cw/edit?tab=t.0).  
				
Also, with Mrinal‘s help I worked up a [draft PR](https://github.com/ga4gh/va-spec/pull/346) defining the classes/attributes needed for a Statement-based representation, along with some data examples.
				 
Note that the proposal re-introduces a modeling pattern from a few years ago for capturing "ancillary information" within a Statement. This pattern goes back to the [ACM days](https://docs.google.com/presentation/d/17qcy_FYNVx6xcD8X-Yk5c_lXVUBBE-Uo16CM4_GdXAA/edit#slide=id.g56b6b3d1f7_0_683), and was originally informed by several use cases from our original landscape analysis. This pattern seemed a perfect fit for capturing some of the information accompanying the central MC annotation in VEP data, and I think will be a generally useful pattern to have in the model. 
				
More info in the doc and PR, and looking forward to discussing this further.

Mrinal-Thomas-Epic committed:
Add schema for simplified variant moe

Mrinal-Thomas-Epic committed:
Add schema for simplified variant moe

Mrinal-Thomas-Epic committed:
Fixes + add tests

Mrinal-Thomas-Epic committed:
Fixes + add tests

mbrush committed:
Update VariantMolecularConcequenceProposition model

Updates to VariantMolecularConcequenceProposition model to reflect decisions on 9-25-25 VA call:

- Subject: MUST be a genomic variant 
- Predicate: hasMolecularConsequence (constant)
- Object: remains the place where we capture the MC terms - but we are dropping the MolecularConsequence class and using a more generic 'Concept or Concept Set' class 
- Qualifiers: separate qualifiers for transcript and protein variant context

mbrush committed:
Update VariantMolecularConcequenceProposition model

Updates to VariantMolecularConcequenceProposition model to reflect decisions on 9-25-25 VA call:

- Subject: MUST be a genomic variant 
- Predicate: hasMolecularConsequence (constant)
- Object: remains the place where we capture the MC terms - but we are dropping the MolecularConsequence class and using a more generic 'Concept or Concept Set' class 
- Qualifiers: separate qualifiers for transcript and protein variant context

mbrush committed:
Adding ConceptorConceptSet and ConceptSet classes

Additions reflect decisions made on 9-25-25 VA Call

ConceptOrConceptSet class: defined as one of MappableConcept (for a single concept) or ConceptSet (necessarily 2 or more concepts) 

I made ConceptOrConceptSet the parent of Condition and Therapeutic, and I made ConceptSet the parent of ConditionSet and TherapyGroup  (@ahwagner  @larrybabb we didn’t explicitly discuss this, but I assumed it was the plan)

mbrush committed:
Adding ConceptorConceptSet and ConceptSet classes

Additions reflect decisions made on 9-25-25 VA Call

ConceptOrConceptSet class: defined as one of MappableConcept (for a single concept) or ConceptSet (necessarily 2 or more concepts) 

I made ConceptOrConceptSet the parent of Condition and Therapeutic, and I made ConceptSet the parent of ConditionSet and TherapyGroup  (@ahwagner  @larrybabb we didn’t explicitly discuss this, but I assumed it was the plan)

mbrush committed:
Added geneConceptQualifier to MC Proposition

This attribute is no longer inherited, as the MC Proposition class now descends directly from SubjectVariantProposition.

mbrush committed:
Added geneConceptQualifier to MC Proposition

This attribute is no longer inherited, as the MC Proposition class now descends directly from SubjectVariantProposition.

mbrush committed:
Improved description of the membershipOperator attribute

mbrush committed:
Improved description of the membershipOperator attribute

mbrush committed:
Update example to align with recent schema changes

mbrush committed:
Update example to align with recent schema changes

mbrush committed:
Update example to align with recent schema changes

mbrush committed:
Update example to align with recent schema changes

mbrush committed:
Update example to align with recent schema changes

mbrush committed:
Update example to align with recent schema changes

mbrush commented:
@Mrinal-Thomas-Epic @ahwagner @larrybabb - I updated this MC Profile PR (schema, and test fixture examples) according to implement decisions from the 9-25-25 VA Call. Please review to make sure I got things right.  

I made a separate PR (#375) to rename ClinicalVariantProposition -> GeneticContextVariantPropositon. 

Below is a summary of these decisions, the corresponding model changes, and some questions:

### va-core-source.yaml
- `subjectVariant`: MUST be a genomic variant 
   - I specified this informally  in the attribute description as a MUST convention  (AFAIK there is no way to formally encode this in the model?) 
   - **Question:** @ahwagner  given this convention, do the four variation classes in the oneOf clause for this attribute make sense?  i.e. does it make sense to allow Categorical variants, or Adjacencies, in the subject field given the requirement to be genomic?
- `objectConsequence`: remains the place where we capture the MC terms - but we are dropping the MolecularConsequence class and using a more generic 'Concept or Concept Set' class (I don’t love the name, but we can change later if we want)
- `qualifiers`: separate, single-valued qualifiers for transcript and protein variant context
   - `transcriptVariationContextQualifier`, and `proteinVariationContextQualifier`
   - **Question:** @ahwagner @larrybabb please review descriptions and informal requirements I added here - which reflect what I understood from our conversations.
	
### domain-entities-source.yaml
- `ConceptOrConceptSet`: 
   - defined as oneOf MappableConcept (for a single concept) or ConceptSet (necessarily 2 or more concepts) 
   - made this the parent of Therapy and Condition - @ahwagner @larrybabb we didn’t explicitly discuss this, but I assumed it was the plan.
   - This parallels pattern for Therapeutic as one of MappableConcept or TherapyGroup, and Condition as one of Mappable:Concept or Conditionset.
   - **Questons**:
       - confirm that "OR" is the proper membershipOperator when more than one MC term apply for the same variant context (e.g. in the intronic-consequence test fixture example. See my rationale of this attribute in the ConceptSet class.
       - given the inheritance pattern for Concept Sets above, the membershipOperator attribute in TherapyGroup and ConditionSet should inherit the enumeration binding from the parent ConceptSet class - so can we remove these from the child classes? Note that the descriptions/comments on these child classes are specialized, however, so keep these as extensions. 

- `ConceptSet`:
   - defined as the parent of TherapyGroup and ConditionSet  - @ahwagner @larrybabb we didn’t explicitly discuss this, but I assumed it was the plan.
	

### Test Fixtures:
- updated to comply with the changes above

### Additional Questions:
   - should we add a proper genomic variant Allele object as the subject  in test fixture examples to illustrate this requirement and data structure
   - Is MC another case like the MAVE EVFI work where we don't define a Statement profile? I understand how this fits with the convention, but FWIW I find the convention unsatisfying. I feel like it a Statement profile for MACE, VEP MC, etc would give something more concrete to point to, that is easier to understand and use, and provides opportunities to tailor the Statement to implementer needs.

larrybabb reviewed (changes requested):
Great job @mbrush ! Overall I like the changes and think they work. I still need to pull this down and run the tests, etc. but I'm staring with this preliminary review.

One topic we should discuss before merging is whether `ConceptSet` belongs in the gks core so that Cat-VRS and VRS can use it if needed. This class seems to be a further wrapping and use of the general use `MappableConcept` class and therefore makes more sense to be in the core library IMO. 

This is potentially a bigger decision and should be thoughtfully considered. Doing this would cause us to have to do a point release for every product (I think), maybe a patch release.

I also want to verify with @ahwagner whether va-spec would need a minor release change or path release change to it's version. Since the ConditionSet is in `trial use` I think this would be a breaking change (i.e. the "conditions" attibritue becomes "concepts".

Mrinal-Thomas-Epic commented:
> * should we add a proper genomic variant Allele object as the subject  in test fixture examples to illustrate this requirement and data structure

I think it would be good to include a full Allele in one of them, just for demonstration purposes. Regarding the genomic subjectVariant requirement, VEP does allow you to provide a c. variant as input, but I believe it converts it to genomic before then calculating consequences across all transcripts ([example](https://rest.ensembl.org/vep/human/hgvs/ENST00000811587.1:n.327-16745dup?content-type=application/json&hgvs=true)). Let's check with @sarahhunt , whether it makes sense to introduce this requirement

> * confirm that "OR" is the proper membershipOperator in the intronic-consequence example (see my description of this attribute in the ConceptSet class description)

I would argue "AND" fits better, since both terms apply.


Super minor, but where did we land on the word `Context` in qualifier names again? Everything else looks good to me!

mbrush commented:
Thanks for the feedback @Mrinal-Thomas-Epic . Addressing your points below:
 
- If others agree that the test fixture data examples should include a proper genomic variant, hoping @larrybabb  or @ahwagner or @korikuzma might add this in. (Or at least a proper va hash id reference for the variant)
- Re: the correct logical operator - the semantics of "OR" is that all members of the set apply on their own, independent of the other (not that one of the set applies but possible not the other).  This is why I think "OR" is correct here.  The "AND" operator means that the members of the set are considered together as a co-occurring collective.  This is applicable for things like therapy sets, or condition sets - but I don’t think makes much sense for MC term sets. 
- Finally, I think we agreed that "context" should remain in the qualifier names as they were implemented in my recent updates to the PR.

mbrush commented:
@larrybabb Thanks for the feedback as well. 

I agree that `ConceptSet` and `ConceptOrConceptSet` may be candidates for gks-core. And that it is not critical to address now as it may require broader conversation, and add more work that would slow things now. Curious to hear from @ahwagner about this, your release question, and the other questions I posed in my [comment above](https://github.com/ga4gh/va-spec/pull/336#issuecomment-3340471197) so we can merge this and be on our way!

Finally, @larrybabb it says you 'requested changes' - but I’m not sure how to tell what these changes are?

ahwagner commented on schema/va-spec/base/domain-entities-source.yaml:
```diff
@@ -10,9 +10,59 @@ namespaces:
+    $comment: >-
+      Individual concepts are represented using the MappableConcept class, which supports optional mappings and metadata
+      about the code system. Sets of concepts are represented using the ConceptSet class, which is an array of 2 or more
+      MappableConcept objects.
```
I do not remember discussing this pattern, and have some concerns about its usage below.

ahwagner commented on schema/va-spec/base/domain-entities-source.yaml:
```diff
@@ -10,9 +10,59 @@ namespaces:
   # Condition data structure for consolidating the different forms of conditions, individual vs co-occurring traits
   Condition:
     maturity: trial use
+    inherits: ConceptOrConceptSet
```
So a Condition is a concept or concept set, and a conditionset is a concept set? This seems wrong to me.

ahwagner commented on schema/va-spec/base/domain-entities-source.yaml:
```diff
@@ -69,8 +122,10 @@ $defs:
+      
   # Therapeutic
   Therapeutic:
+    inherits: ConceptOrConceptSet
```
Same concerns here as with condition.

ahwagner reviewed (changes requested):
The SPOQ properties look pretty good to me. Agree with @larrybabb re: moving the conceptset to GKS Core.

I have some reservations about the new "ConceptOrConceptSet" class introduced here, and its usage.

Changing to concept set will be a breaking change of a trial use class, and therefore necessitate a minor version increment, which should be included in this PR.

mbrush committed:
Update Condition and Therapeutic entities to inherit from MappableConcept

- Removed ConceptOrConceptSet class
- Changed definition of Condition to be a single condition, and made it inherit from Mappable Concept, with appropriate adjustments to accommodate this
- Changed definition of Therapeutic to be a single therapy, and made it inherit from Mappable Concept, with appropriate adjustments to accommodate this (also changed name to “Therapy”, for consistency)

mbrush committed:
Update Condition and Therapeutic entities to inherit from MappableConcept

- Removed ConceptOrConceptSet class
- Changed definition of Condition to be a single condition, and made it inherit from Mappable Concept, with appropriate adjustments to accommodate this
- Changed definition of Therapeutic to be a single therapy, and made it inherit from Mappable Concept, with appropriate adjustments to accommodate this (also changed name to “Therapy”, for consistency)

mbrush committed:
Changes to accommodate remodeling of Concepts,  Conditions and Therapies

Changes to accommodate remodeling of Conditions and Therapies in domain-entities.yaml:
- Changed “Therapeutic” -> “Therapy” where needed to match change in the name of the “Therapeutic” class to “Therapy”
- Changed references to Condition to “Condition” or “Condition Set” to accommodate changes to Condition modeling.

Also removed  “- $refCurie: cat-vrs:CategoricalVariant” as an option for the MC proposition subjectVariant

mbrush committed:
Changes to accommodate remodeling of Concepts,  Conditions and Therapies

Changes to accommodate remodeling of Conditions and Therapies in domain-entities.yaml:
- Changed “Therapeutic” -> “Therapy” where needed to match change in the name of the “Therapeutic” class to “Therapy”
- Changed references to Condition to “Condition” or “Condition Set” to accommodate changes to Condition modeling.

Also removed  “- $refCurie: cat-vrs:CategoricalVariant” as an option for the MC proposition subjectVariant

mbrush commented:
Made changes to address outcomes of Oct 22 VA Call.

@ahwagner @larrybabb please closely check the way I implemented this to make sure it aligns with conventions and expectations. 

**domain-entities.yaml changes:**
- Removed `ConceptOrConceptSet` class
- Changed definition of `Condition` and `Therapeutic` classes to refer to a single condition/therapy, and made them inherit from `MappableConcept` (with appropriate adjustments to the classes to accommodate this)
- Also changed name of "Therapeutic" class to “Therapy”, for consistency.


**va-core-source.yaml changes**: (to accommodate changes above)
- Changed references to `Condition`  to now refer to  oneOf  `Condition` or `Condition Set` - to accommodate changes to Condition class described above.
- Changed text references to “Therapeutic” --> “Therapy” where needed to match change in the name of the `Therapeutic` class to 
“Therapy”
- Unrelated: I removed  `cat-vrs:CategoricalVariant”` as an option for the MC proposition `subjectVariant` attribute

**Questions:** 
- We fundamentally changed the definition of the `Condition` and `Therapy`  classes, which was trial use.  If this is a breaking change, consider making a minor version release?
- I made `conceptType`  a required attribute in `Condition` and `Therapy` classes – since we are constraining it to have a single specified value (“Condition”).  Is this ok?

Mrinal-Thomas-Epic committed:
Add genomic subject variant to fixtures

Mrinal-Thomas-Epic committed:
Add genomic subject variant to fixtures

Mrinal-Thomas-Epic committed:
Add breakpoint examples

Mrinal-Thomas-Epic committed:
Add breakpoint examples

Mrinal-Thomas-Epic commented:
This may end up being implementation-specific, but wanted to mention it here for visibility. For VEP, we may want to bundle together multiple statements in the output, grouping by the `subjectVariant` or its `location` (in the case of multiple variants at one location).

Another question that was brought up was grouping by location might become slightly complicated if VCF was the input and the location of one variant changed due to normalization, but the other didn't. If both variants are insertions or deletions, this shouldn't be a concern. However, if one variant is a substitution and the other is an insertion or deletion, we might run into issues

Mrinal-Thomas-Epic committed:
Fix build

Mrinal-Thomas-Epic committed:
Fix build

larrybabb commented:
We need to move `ConceptSet` to gks-core. We can do this after we get the "breaking changes" approved for `ConditionSet.concepts` and `TherapyGroup.concepts`.

Mrinal-Thomas-Epic committed:
Allow direct re-use of ConceptSet

Mrinal-Thomas-Epic committed:
Allow direct re-use of ConceptSet

Mrinal-Thomas-Epic committed:
fix examples

Mrinal-Thomas-Epic committed:
fix examples

Mrinal-Thomas-Epic committed:
Fix tests

Mrinal-Thomas-Epic committed:
Fix tests

korikuzma commented on schema/va-spec/base/domain-entities-source.yaml:
Out of curiosity, why are we setting `additionalProperties: true`? It seems like we usually only do this when the type is an object. Otherwise, we recommend people to store additional properties as `Extension`s. Just wanted to see if I'm missing something

Mrinal-Thomas-Epic replied:
Oh I just realized `additionalProperties` defaults to true without me explicitly specifying it so its not needed.

The reason I added it (and we need to keep using the default) is that the two schemas in `anyOf` are validated independently. If I set `additionalProperties: false` in the second schema (where I'm validating `Concepts` are `Condition` or `ConditionSet`), I would need to specify all the other `ConceptSet` properties in the second schema (`membershipOperator`, etc.).

The first schema in the `anyOf` doesn't allow `additionalProperties`, so implementers can't actually specify arbitrary properties not in the schema

korikuzma replied:
Thanks @Mrinal-Thomas-Epic !

korikuzma reviewed (changes requested):
Looks like there's some merge conflicts to resolve.

Also minor question

Mrinal-Thomas-Epic committed:
Remove unneeded additionalProperties

Mrinal-Thomas-Epic committed:
Remove unneeded additionalProperties

larrybabb reviewed (commented):
@ahwagner @Mrinal-Thomas-Epic 
I'm wondering if we should move `ConceptSet` to `gks-core` now or wait. I think this will be something that cat-vrs eventually needs, so maybe we wait until there's evidence. The `ConceptSet` is very general so I believe it really belongs in gks-core.  I think it will ultimately end up there, but I'm not certain if it would be a major or minor breaking change if we decide to move it later.  
This is probably a bigger general question about all the domain concepts unless we get by with using `MappableConcept` for any things that are concept-like in Cat-VRS and VRS (i.e. Gene).

Mrinal-Thomas-Epic commented:
> @ahwagner @Mrinal-Thomas-Epic I'm wondering if we should move `ConceptSet` to `gks-core` now or wait. I think this will be something that cat-vrs eventually needs, so maybe we wait until there's evidence. The `ConceptSet` is very general so I believe it really belongs in gks-core. I think it will ultimately end up there, but I'm not certain if it would be a major or minor breaking change if we decide to move it later. This is probably a bigger general question about all the domain concepts unless we get by with using `MappableConcept` for any things that are concept-like in Cat-VRS and VRS (i.e. Gene).

I agree with moving `ConceptSet` to `gks-core` now. I just figured it would be easier for review to view it all on one repo. I can move it before merging

likhitha-surapaneni commented on tests/fixtures/vep-transcript-proposition.yaml:
```diff
@@ -0,0 +1,75 @@
+      residueAlphabet: aa
+  state:
+    type: LiteralSequenceExpression
+    sequence: V
```
Hi @Mrinal-Thomas-Epic , thank you for this. 

I am looking at the following examples and wondering if the following cases are within the scope of the current model (and accounted for) :

-  frameshift: [NC_000023.11:g.77988720_77988721del](https://rest.ensembl.org/vep/human/hgvs/NC_000023.11:g.77988720_77988721del?content-type=application/json) where amino acid is returned as `amino_acids":"Q/X"`
- splice_donor_variant: [NC_000023.11:g.78013111_78013113delinsT](https://rest.ensembl.org/vep/human/hgvs/NC_000023.11:g.78013111_78013113delinsT?content-type=application/json) where `amino_acids` are not returned
- stop_lost: [NC_000024.10:g.334343_334379delT](https://rest.ensembl.org/vep/human/hgvs/NC_000024.10:g.334343_334379delT?content-type=application/json) where `amino_acids` are not returned
- start_lost: [NC_000013.11:g.34942816_34942874del](https://rest.ensembl.org/vep/human/hgvs/NC_000013.11:g.34942816_34942874del?content-type=application/json) where `protein_start` and `amino_acids` are not returned

Mrinal-Thomas-Epic replied:
Hi Likhitha,

Great question! Right now, those variants cannot cleanly be expressed in VRS (unless you are put the entire resulting protein sequence as an Allele). I think in the short term, we wouldn't include a `proteinVariationContextQualifier` for any of those cases (frameshift, splice_donor_variant, stop_lost, start_lost), but could potentially add extensions for discrete data related to those. We could still include the transcript change representation and molecular consequence terms.

In the long run, we would add support for these variant types in VRS, and then the VRS representation could be used in this slot.

@ahwagner , @larrybabb , is this your understanding as well?

larrybabb replied:
That makes sense to me, but I'll defer to @ahwagner for the expert final say.  I do think this should be added as a git issue in this repo so we don't lose track of it going forward. @likhitha-surapaneni do you mind adding your request as a git issue once @ahwagner weighs in with his thoughts?

sarahhunt commented on schema/va-spec/base/def/VariantMolecularConsequenceProposition.rst:
```diff
@@ -0,0 +1,89 @@
+
+**Computational Definition**
+
+A Proposition describing a type of consequence of a variant on transcript and protein molecules - typically reporting the type of sequence feature affected (e.g. 'intron variant', 'splice-site variant'), or an impact on the processing of the molecule along the path from gene to transcript to polypeptide (e.g. 'missense variant', 'frameshift variant'). Note that annotations about variant impact on ...
```
'consequence of a variant' -> 'predicted molecular consequence of a variant'

Mrinal-Thomas-Epic replied:
Updated!

sarahhunt commented on schema/va-spec/base/def/VariantMolecularConsequenceProposition.rst:
```diff
@@ -0,0 +1,89 @@
+      -
+      - :ref:`MappableConcept` | :ref:`iriReference`
+      - 0..1
+      - Reports a gene impacted by the variant, which may contribute to the association described in the Proposition.
```
'impacted' suggests strongly affected, but we will also use this for variant consequence types which may not  necessarily have an effect on the gene ( eg. synonymous, intronic).
Something like 'The gene for which this statement reports a VariantMolecularConsequence' would be clearer.

Mrinal-Thomas-Epic replied:
Updated!

DanielPuthawala commented:
For the record, though I don't have a use case ready on the top of my brain for `ConceptSet` in Cat-VRS, I think `ConceptSet` is very nifty and general-purpose and support that being in `gks-core` so we can reuse it. (And I realize that move already happened.)

larrybabb commented:
FYI - @korikuzma is in the process of moving the `ConceptSet` to gks-core and updating the submodules and removing it from va-spec, so that change is forthcoming.

Mrinal-Thomas-Epic committed:
Update descriptions for VariantMolecularConsequenceProposition and geneContextQualifier

Mrinal-Thomas-Epic committed:
Update descriptions for VariantMolecularConsequenceProposition and geneContextQualifier

Mrinal-Thomas-Epic commented:
Note: need to fix [Condition.rst](https://github.com/ga4gh/va-spec/blob/104e8bcd005dafa3330b72ef023fa5fc235e904e/schema/va-spec/base/def/Condition.rst).

See related discussion in [gks-core](https://github.com/ga4gh/gks-core/discussions/127#discussioncomment-15397507)

Mrinal-Thomas-Epic committed:
Use ConceptSet from gks-core

Mrinal-Thomas-Epic committed:
Use ConceptSet from gks-core

Mrinal-Thomas-Epic committed:
fix tests

Mrinal-Thomas-Epic committed:
fix tests

Mrinal-Thomas-Epic committed:
Update docs

Mrinal-Thomas-Epic committed:
Update docs

Mrinal-Thomas-Epic committed:
Update docs for Condition, ConditionSet, Therapy, TherapySet

Also rebase fixes

Mrinal-Thomas-Epic committed:
Update docs for Condition, ConditionSet, Therapy, TherapySet

korikuzma reviewed (dismissed):
Should pull in latest changes from v1

korikuzma committed:
Merge remote-tracking branch 'origin/v1' into epic-vep-molecular-consequence

larrybabb reviewed (approved):
pre-review passed. ready for public review
